File: task.swift
import Foundation

func compress(_ s: String) -> String {
    var result = ""
    var count = 1
    let chars = Array(s)
    
    for i in 0..<chars.count {
        if i + 1 < chars.count && chars[i] == chars[i + 1] {
            count += 1
        } else {
            result += "\(chars[i])\(count)"
            count = 1
        }
    }
    
    return result.count < s.count ? result : s
}

print("Введите строку: ", terminator: "")
if let input = readLine() {
    print(compress(input))
}

File: task.go
package main

import (
    "fmt"
    "strconv"
)

func compress(s string) string {
    result := ""
    count := 1

    for i := 0; i < len(s); i++ {
        if i+1 < len(s) && s[i] == s[i+1] {
            count++
        } else {
            result += string(s[i]) + strconv.Itoa(count)
            count = 1
        }
    }

    if len(result) < len(s) {
        return result
    }
    return s
}

func main() {
    var input string
    fmt.Print("Введите строку: ")
    fmt.Scan(&input)
    fmt.Println(compress(input))
}

